### DISCUSSION TITLE
There is a detached mode like Docker ?

### DISCUSSION THREAD

**Discussion by valmnt** [cirruslabs/tart#449] [Q&A] [answered]
Hi,

I don't think Tart has a detached mode like docker does with the parameter "-d".

So i was wondering if you have any plans for this feature ? 

Thx for reading,
Valmnt

**Comment by fkorotkov** [comment, marked as answer]
There is no detach mode. We'll try to have something similar for Desktop Application (#305). At the moment the CLI usage is designed to be daemon-less so consumers are responsible for cleaning VMs. You can workaround it with `nohup` though:

```console
nohup tart run --no-graphics <VM-name> &
```

The only cons is that if your VM fails to start you won't know it. But you can also workaround it by checking JSON output of `tart get` and looking at `Running` field:

```console
$: tart get --format=json base
{
  "Memory" : 8192,
  "CPU" : 4,
  "Display" : "1024x768",
  "Disk" : 19,
  "Running" : true
}
```

**Comment by fabiannagel** [comment]
IMHO this is a bit clunky. I'd prefer not having to open up another terminal instance to SSH into the machine. Also, by now people are probably used to the way Docker is handling this.

**Comment by pauldsmyth** [comment]
I'm interested in the potential for this, mainly because I'd like to be able to cleanly start and stop the VM automatically on startup/shutdown so that in the event that my UPS reports a power outage I'm able to cleanly shutdown services and restore them without manual intervention if possible.

**Reply by edigaryev** [reply]
Have you considered using [launchd](https://launchd.info/) for that purpose?

See https://github.com/cirruslabs/tart/issues/374#issuecomment-1536518622 for more details.

**Reply by pauldsmyth** [reply]
Oh that looks interesting. Thanks I'll try that
